flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -740,49 +711,39 @@ public void testFailureWhileRestarting() throws Exception {
 	//  Utilities
 	// ------------------------------------------------------------------------
 
-	private Scheduler createSchedulerWithInstances(int num, TaskManagerGateway taskManagerGateway) {
+	private SlotProvider createSchedulerWithInstances(JobID jobID, int slots) {
```

Comment:
@tillrohrmann The last usage of Instance in valid tests is in this method. I have no idea how to get rid of it. Could you take a look or give some advice? Thanks!

Reply:
FYI, I break this pr into separated pr. But this help request is still valid :-)